service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerAboutToSubmitTest.java

```diff
@@ -128,6 +132,47 @@ void aboutToSubmitShouldNotHaveDraftAppendedToFilename() {
         assertThat(caseData.getOrderCollection().get(0).getValue().getDocument()).isEqualTo(expectedDocument());
     }
 
+    @Test
+    void shouldAddCloseCaseObjectToCaseDataWhenOptionWasSelectedAndChangeState() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = buildJudgeAndLegalAdvisor(NO);
+
+        final CaseDetails caseDetails = buildCaseDetails(
+            commonCaseDetailsComponents(CARE_ORDER, FINAL, judgeAndLegalAdvisor)
+                .caseLocalAuthority(LOCAL_AUTHORITY_CODE)
```

Comment:
just a heads up from my PR, I put the .caseLocalAuthority inside commonCaseDetailsComponents because it is a common casedetails component, so if mine gets merged before yours make sure you handle that!